src/nvim/message.c

```diff
@@ -264,6 +264,10 @@ bool msg_attr_keep(char_u *s, int attr, bool keep, bool multiline)
   int retval;
   char_u *buf = NULL;
 
+  if (quit_more) {
```

Comment:
don't we still want to add it to message history if applicable?

Reply:
no. we did not bother reading long output till the end and pressed `q`